source/common/http/http1/codec_impl.cc

```diff
@@ -471,11 +478,13 @@ void ServerConnectionImpl::onResetStream(StreamResetReason reason) {
 
 void ServerConnectionImpl::sendProtocolError() {
   // We do this here because we may get a protocol error before we have a logical stream. Higher
-  // layers can only operate on streams, so there is no coherent way to allow them to send a 400
+  // layers can only operate on streams, so there is no coherent way to allow them to send an error
   // "out of band." On one hand this is kind of a hack but on the other hand it normalizes HTTP/1.1
   // to look more like HTTP/2 to higher layers.
   if (!active_request_ || !active_request_->response_encoder_.startedResponse()) {
-    Buffer::OwnedImpl bad_request_response("HTTP/1.1 400 Bad Request\r\n"
+    Buffer::OwnedImpl bad_request_response("HTTP/1.1 " + std::to_string(enumToInt(error_code_)) +
```

Comment:
oh awesome.  I'd been trying to find the envoy equivalent of strcat and failing.  Fixed!